contracts/src/primitive/AllowanceMap.sol

```diff
@@ -0,0 +1,47 @@
+/// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
+
+using AllowanceLib for AllowanceMap global;
```

Comment:
can we avoid using global and instead declare it in the RiverPoints contract directly? If we have different versions of AllowanceLib in different facets, can this cause conflicts?

It makes the dependencies more explicit, which is crucial for diamond proxy patterns.